girder/girder_large_image/web_client/vue/components/CompositeLayers.vue

```diff
@@ -108,36 +108,39 @@ export default {
                     enabled: true,
                     min: undefined,
                     max: undefined,
-                    autoRange: undefined
+                    autoRange: undefined,
+                    palette: '#ffffff'  // default color is white
```

Comment:
Won't this make ALL bands white?

Reply:
My intention was to make white the default which is overridden in the case that `this.layers.length > 1`. But I didn't notice the `if (!this.compositeLayerInfo[layerName].palette)` statement which would prevent color assignment. [9ab1a65](https://github.com/girder/large_image/pull/1300/commits/9ab1a65daa8a85267af6319fe2e4f86b05b49376) removes that line. Now colors are properly assigned when there is more than one layer.

Reply:
What if a layer should have white?  Like a brightfield channel?

Reply:
In a quick test, the default colors are no longer being kept (white for brightfield, blue for dapi, etc.)

Reply:
Ok, thanks for catching that. It's been a while since I've looked at the color assignment code, but I updated it in [742af06](https://github.com/girder/large_image/pull/1300/commits/742af068f98a912c8a1598f716ba4592328e600b) to account for the white default and still do color assignment like normal when there is more than one layer. Could you test again?

Reply:
It all appears to work in my quick tests.  Would it have been better to have just tried to use the band names for the color lookup, since in a single band is called "gray"?

Reply:
In thinking that through, we have some geospatial sources that could expose named bands with some missing (e.g., a three-band image with interpretations of red, 0, 0 (where the other two are near-ir bands).  I'll have to dig around to find such a file.

Reply:
I can do that, too. I would just need to add a new mapping to `colors.js`. In the case you mentioned, though, wouldn't it be better to assign colors based on index rather than name?

Reply:
Superficially, it looks like if you reverted to a649a88 and added 
```
    '^red$': '#FF0000',
    '^green$': '#00FF00',
    '^blue$': '#0000FF',
    '^gr[ae]y(|scale)$': '#FFFFFF',
```
to the color table, it would do what we want.

As far as I can tell, we don't get useful band interpretations unless they are red/green/blue/gray/alpha; otherwise it is either "palette" or "other" (also called 0).  If we have the useful band name, then the color table works.

Reply:
In the red/near-ir example, they could theoretically be stored in a different order (so you'd get [0, 0, red] as the interpretations), so we would still want to look up the color by interpretation name.

Reply:
Ok, I rewrote the history with that solution instead: [476288d](https://github.com/girder/large_image/pull/1300/commits/476288dfb54d8c6f9f40af3f10d9ecb0a587aaaa) Thanks!